include/nil/crypto3/zk/snark/systems/plonk/placeholder/permutation_argument.hpp

```diff
@@ -154,11 +158,26 @@ namespace nil {
                         math::polynomial_dfs<typename FieldType::value_type> V_P_shifted =
                             math::polynomial_shift(V_P, 1, basic_domain->m);
 
-                        F_dfs[0] = preprocessed_data.common_data.lagrange_0 * (one_polynomial - V_P);
-                        F_dfs[1] = (one_polynomial - (preprocessed_data.q_last + preprocessed_data.q_blind)) * (V_P_shifted * h - V_P * g);
-                        F_dfs[2] = preprocessed_data.q_last * V_P * (V_P - one_polynomial);
+                        F_dfs[0] = one_polynomial;
+                        F_dfs[0] -= V_P;
+                        F_dfs[0] *= preprocessed_data.common_data.lagrange_0;
+
+                        math::polynomial_dfs<typename FieldType::value_type> t1 = V_P;

```

Comment:
Same here ,let's keep the initial code as comment for better readability.

Reply:
updated in 9866f9a2